### DISCUSSION TITLE
Is this the correct way to use byte::Regex?

### DISCUSSION THREAD

**Discussion by kirs112** [rust-lang/regex#1246] [Q&A] [answered]
I am refactoring my python code, but there is a problem below. When refactoring using rust, it always fails to match. Can you help me point out where the problem is?

python code
```
import re

date = b'\x04\x01\x00%\x00\x00\x01\x00\x00\x00\x15\x00\x06\x01\x00\x1b\x00\x01\x02\x00\x1c\x00\x01\x03\x00\x1d\x00\x00\xff\x0f\x00\x109\x00\x00\x00\x00'
pattern = b'^\x04\x01\x00\x25\x00\x00\x01\x00\x00\x00\x15\x00\x06\x01\x00\x1b\x00\x01\x02\x00\x1c\x00\x01\x03\x00\x1d\x00\x00\xff\x0f\x00(..)'

# Use bytes pattern and data
pattern_compiled = re.compile(pattern, re.DOTALL)
rfind = pattern_compiled.findall(date)

print(rfind)  # Expected: [b'\x10\x39']

```
rust code
```
use regex::bytes::Regex;

fn main() {
    // Define the byte data
    let date = b"\x04\x01\x00\x25\x00\x00\x01\x00\x00\x00\x15\x00\x06\x01\x00\x1b\x00\x01\x02\x00\x1c\x00\x01\x03\x00\x1d\x00\x00\xff\x0f\x00\x10\x39\x00\x00\x00\x00";
    
    // Define the pattern
    let pattern = r"^\x04\x01\x00\x25\x00\x00\x01\x00\x00\x00\x15\x00\x06\x01\x00\x1b\x00\x01\x02\x00\x1c\x00\x01\x03\x00\x1d\x00\x00\xff\x0f\x00(..)";
    
    // Create regex pattern
    let re = Regex::new(pattern).unwrap();
    
    // Find all matches
    let matches: Vec<Vec<u8>> = re.captures_iter(date)
        .filter_map(|cap| cap.get(1))
        .map(|m| m.as_bytes().to_vec())
        .collect();
    
    println!("Matches: {:?}", matches);  // Should print: [[16, 57]] (which is [\x10\x39] in hex)
}

```

**Comment by BurntSushi** [comment, marked as answer]
You almost got it. You just need to disable Unicode mode in your pattern. e.g., `(?-u)^blahblah`. Otherwise, things like `\xFF` are interpreted as the "the UTF-8 encoding of `U+00FF`," which for non-ASCII codepoints, is different than the raw byte `\xFF`. By disabling Unicode mode, `\xFF` is interpreted as matching the raw byte `\xFF`.

**Reply by BurntSushi** [reply]
This is documented here: https://docs.rs/regex/latest/regex/bytes/index.html#syntax

And an example here: https://docs.rs/regex/latest/regex/bytes/index.html#example-selectively-enable-unicode-support

And here: https://docs.rs/regex/latest/regex/bytes/index.html#example-match-null-terminated-string
